lua/cfc_random_spawn/module/sv_player_spawning.lua

```diff
@@ -9,23 +10,76 @@ CFCRandomSpawn.spawnPointRankings = CFCRandomSpawn.spawnPointRankings or {}
 
 local function getMeasurablePlayers()
     local measurablePlayers = {}
-    for _, ply in pairs( player.GetHumans() ) do
-        if ( ply:Alive() ) then
+    for _, ply in pairs( player.GetAll() ) do
+        if ply:Alive() and not ply.isCFCRespawning then
             table.insert( measurablePlayers, ply )
         end
     end
 
     return measurablePlayers
 end
 
+
+-- Only use spawnpoints that are nearest to the most "popular" point
+-- This lets everyone spawn near the most active part of the server
+
+-- if count is hit and distance is not hit, more spawnpoints will be chosen & vice versa
+
+local nearSpawnpointsMinCount = 8 -- minimum count of spawnpoints that will be availiable 
+local nearSpawnpointsMinDistance = 2500 -- minimum size of area that spawnpoints will be chosen from
+
+local function sortSpawnsByDistance( comparePos, spawns ) 
+    table.sort( spawns, function( a, b ) -- sort by distance
+        local aDist = a.spawnPos:DistToSqr( comparePos )
+        local bDist = b.spawnPos:DistToSqr( comparePos )
+        return aDist < bDist 
+    end )
+    return spawns
+end
+
+local function getNearestSpawns( nearPos, spawns, minExtent, minDist )
+    local distSortedSpawns = sortSpawnsByDistance( nearPos, spawns )
+    local bestSpawn = distSortedSpawns[1] -- get best spawn so the distance comparison isnt worthless
+    local operationCount = 0
+    local nearestSpawns = {}
+
+    for _, spawn in pairs( distSortedSpawns ) do
+        operationCount = operationCount + 1
+        local distToFirst = bestSpawn.spawnPos:Distance( spawn.spawnPos ) -- will never run on every spawnpoint
+        local overCount = operationCount >= minExtent
+        local overDistance = distToFirst > minDist
+        if overCount and overDistance then break
+        else 
+            nearestSpawns[_] = spawn
+        end
+    end
+    return nearestSpawns
+end
+
+local function getPopularPoint( players )
+    if players == nil then return Vector( 0 ) end
+    local average = Vector( 0 )
+    local playersCount = table.Count( players )
+    for _, currentPlayer in pairs( players ) do
+        if IsValid( currentPlayer ) then
+            average = average + currentPlayer:GetPos()
+        end
+    end
+    average = average / playersCount
+    
+    debugoverlay.Cross( average, 200, 100, Color( 255, 255, 255 ), true )
```

Comment:
Might wanna remove this, even if it's a debugoverlay thing it *might* still network stuff to players, could just comment it out for later use if needed.